lib/pure/unicode.nim

```diff
@@ -1010,28 +1001,15 @@ proc splitWhitespace*(s: string): seq[string] {.noSideEffect,
 
 iterator split*(s: string, sep: Rune, maxsplit: int = -1): string =
   ## Splits the unicode string ``s`` into substrings using a single separator.
-  ##
   ## Substrings are separated by the rune ``sep``.
-  ## The code:
-  ##
-  ## .. code-block:: nim
-  ##   for word in split(";;this;is;an;;example;;;", ';'):
-  ##     writeLine(stdout, word)
-  ##
-  ## Results in:
-  ##
-  ## .. code-block::
-  ##   ""
-  ##   ""
-  ##   "this"
-  ##   "is"
-  ##   "an"
-  ##   ""
-  ##   "example"
-  ##   ""
-  ##   ""
-  ##   ""
-  ##
+  runnableExamples:
+    var splitted = newSeq[string]()
+
+    for word in split(";;this;is;an;;example;;;", ";".runeAt(0)):
+      splitted.add(word)
+
+    assert splitted == @["", "", "this", "is", "an", "", "example", "", "", ""]
+
```

Comment:
```suggestion
    import sequtils
    assert toSeq(";;this;is;an;;example;;;", ";".runeAt(0)) == @["", "", "this", "is", "an", "", "example", "", "", ""]
```

and also, use unicode in the input struct instead of plain ascii otherwise user might as well use strutils.split

Reply:
`toSeq(";;this;is;an;;example;;;", ";".runeAt(0))` doesn't entirely work? Where do I call the `split` function in this line? 
This compiles:
```nim
import std/sequtils

assert toSeq(split(";;this;is;an;;example;;;", ";".runeAt(0))) ==
  @["", "", "this", "is", "an", "", "example", "", "", ""]

``` 
but it's getting really hairy to read personally haha.
Also do you have an example of what you mean example with use "use unicode in the input struct instead of plain ascii" ? Do you mean a string containing unicode?

Reply:
right, i forgot `split`.
> but it's getting really hairy 

i think it's fine, and it's also idiomatic; but feel free to remove whatever you deem redundant, eg:

```nim
assert toSeq(split(";;ab;是;å;;de;;", ";".runeAt(0))) == ...
```
> Do you mean a string containing unicode?

yes, see above